circuits/cpp/src/aztec3/circuits/kernel/private/.test.cpp

```diff
@@ -29,89 +28,38 @@ class private_kernel_tests : public ::testing::Test {
+ * @details The purpose of this test is to check the private kernel circuit given a valid app proof and a valid previous
+ * private kernal proof. To avoid doing actual proof construction, we simply read in an arbitrary but valid proof and a
+ * corresponding valid verification key from file. The same proof and vkey data is used for both the app and the
+ * previous kernel.
+ * @note The choice of app circuit (currently 'deposit') is entirely arbitrary and can be replaced with any other valid
+ * app circuit.
  */
-TEST_F(private_kernel_tests, circuit_deposit)
+TEST_F(private_kernel_tests, basic)
 {
     NT::fr const& amount = 5;
     NT::fr const& asset_id = 1;
     NT::fr const& memo = 999;
     std::array<NT::fr, 2> const& encrypted_logs_hash = { NT::fr(16), NT::fr(69) };
     NT::fr const& encrypted_log_preimages_length = NT::fr(100);
 
+    // Generate private inputs including proofs and vkeys for app circuit and previous kernel
     auto const& private_inputs = do_private_call_get_kernel_inputs_inner(
         false, deposit, { amount, asset_id, memo }, encrypted_logs_hash, encrypted_log_preimages_length, true);
 
     // Execute and prove the first kernel iteration
     Composer private_kernel_composer("../barretenberg/cpp/srs_db/ignition");
     auto const& public_inputs = private_kernel_circuit(private_kernel_composer, private_inputs, true);
 
-    // TODO(jeanmon): this is a temporary hack until we have private_kernel_circuit init and inner
-    // variant. Once this is supported, we will be able to generate public_inputs with
-    // a call to private_kernel_circuit_init(private_inputs_init, ...)
-    auto const& private_inputs_init = do_private_call_get_kernel_inputs_init(
-        false, deposit, { amount, asset_id, memo }, encrypted_logs_hash, encrypted_log_preimages_length, true);
-
-    // TODO(jeanmon): Once we have an inner/init private kernel circuit,
-    // there should not be any new deployed contract address in public_inputs
-    // and the following assertion can be uncommented:
-    // validate_no_new_deployed_contract(public_inputs);
-
-    // TODO(jeanmon): Remove once we have an inner/innit private kernel circuit
-    // Check contract address was correctly computed by the circuit
-    EXPECT_TRUE(validate_deployed_contract_address(private_inputs_init, public_inputs));
-    EXPECT_FALSE(private_kernel_composer.failed());
-
-    // Create the final kernel proof and verify it natively.
-    auto final_kernel_prover = private_kernel_composer.create_prover();
-    auto const& final_kernel_proof = final_kernel_prover.construct_proof();
-
-    auto final_kernel_verifier = private_kernel_composer.create_verifier();
-    auto const& final_result = final_kernel_verifier.verify_proof(final_kernel_proof);
-    EXPECT_EQ(final_result, true);
-}
-
-/**
- * @brief Some private circuit proof (`constructor`, in this case)
- */
-TEST_F(private_kernel_tests, circuit_basic_contract_deployment)
```

Comment:
This test was identical to the previous test but for a different "app" circuit (`constructor` instead of `deposit`). Since this test suite is intended to test the private kernel circuit, not the app circuits, this test adds no value, particularly now that we're using an arbitrary valid proof for the internal circuits.